### PR TITLE
end2endtest account: fixes

### PR DESCRIPTION
**Pull request by mariajdab**
Remove `log.Fatal` use and check concurrency the balance/nonce and token transfer count

### CODE REVIEW

**Comment by mariajdab** [comment]
could you please take a look @altergui? The error said `Process completed with exit code 2.` but there is not more information

File: cmd/end2endtest/account.go [hunk 1]
```diff
@@ -169,73 +169,121 @@ func testSendTokens(api *apiclient.HTTPclient, aliceKeys, bobKeys *ethereum.Sign
...
+		}
+		wgCheck.Done()
+	}()
+
+	go func() {
+		if err := checkTokenTransfersCount(alice, aliceKeys.Address()); err != nil {
+			errorChan <- fmt.Errorf("failed when checkTokenTransfersCount for alice: %v", err)
+		}
+		if err := checkTokenTransfersCount(bob, bobKeys.Address()); err != nil {
+			errorChan <- fmt.Errorf("failed when checkTokenTransfersCount for for: %v", err)
```
Comment (altergui) [inline cmd/end2endtest/account.go]: for bob

File: cmd/end2endtest/account.go [hunk 2]
```diff
@@ -169,73 +169,121 @@ func testSendTokens(api *apiclient.HTTPclient, aliceKeys, bobKeys *ethereum.Sign
...
+		}
 	}
-	log.Debugf("mined, tx refs are %+v and %+v", txrefa, txrefb)
 
 	// after a tx is mined in a block, the indexer takes some time to update the balances
 	// (i.e. seconds, if there are votes to be indexed)
 	// give it one more block time
 	_ = api.WaitUntilNextBlock()
 
+	var wgCheck sync.WaitGroup
```
Comment (altergui) [inline cmd/end2endtest/account.go]: i believe you can just reuse wg

**Review by altergui — APPROVED** [review]
reading the diff, LGTM. i'll try to reproduce the error you mention

**Comment by altergui** [comment]
miner2 and miner3 panic'ed at the same time. output is very confusing since it's intertwined

https://github.com/vocdoni/vocdoni-node/actions/runs/7000208867/job/19040533596#step:4:2778

```
panic: sync: negative WaitGroup counter

goroutine 133 [running]:
sync.(*WaitGroup).Add(0xc000483770, 0xffffffffffffffff)
	sync/waitgroup.go:62 +0x1cd
sync.(*WaitGroup).Done(0x473be2ac281f27a8?)
	sync/waitgroup.go:87 +0x2e
main.testSendTokens.func5()
	go.vocdoni.io/dvote/cmd/end2endtest/account.go:277 +0x3d6
created by main.testSendTokens in goroutine 1
	go.vocdoni.io/dvote/cmd/end2endtest/account.go:270 +0x1bd6
```

```
testsuite_7000208867-miner3-1    |   go.vocdoni.io/dvote/tree.(*Tree).Get()
testsuite_7000208867-miner3-1    |       go.vocdoni.io/dvote/tree/tree.go:70 +0x144
testsuite_7000208867-miner2-1    |   github.com/cockroachdb/pebble.(*Iterator).First()
testsuite_7000208867-miner0-1    | 2023-11-27T03:26:57.661Z INF vochain/app.go:198 > commit block height=41 txs=12
testsuite_7000208867-miner2-1    |       github.com/cockroachdb/pebble@v0.0.0-20230928194634-aa077af62593/iterator.go:1605 +0x33d
testsuite_7000208867-miner0-1    | 2023-11-27T03:26:57.673Z INF comet: committed state block_app_hash=F45155B4901A275CF09CDEA966DE6353F3BAC62999BF23A5A2C74F0C3A6D2CCD height=41
testsuite_7000208867-miner2-1    |   github.com/cockroachdb/pebble.(*DB).getInternal()
testsuite_7000208867-miner0-1    | 2023-11-27T03:26:57.684Z INF comet: indexed block events height=41
testsuite_7000208867-miner2-1    |       github.com/cockroachdb/pebble@v0.0.0-20230928194634-aa077af62593/db.go:595 +0x6c9
testsuite_7000208867-miner2-1    |   github.com/cockroachdb/pebble.(*Batch).Get()
testsuite_7000208867-miner3-1    |   go.vocdoni.io/dvote/statedb.(*TreeUpdate).Get()
testsuite_7000208867-miner2-1    |       github.com/cockroachdb/pebble@v0.0.0-20230928194634-aa077af62593/batch.go:553 +0xa4
testsuite_7000208867-miner3-1    |       go.vocdoni.io/dvote/statedb/treeupdate.go:63 +0x207
testsuite_7000208867-miner2-1    |   go.vocdoni.io/dvote/db/pebbledb.get()
testsuite_7000208867-miner3-1    |   go.vocdoni.io/dvote/statedb.(*TreeUpdate).DeepGet()
testsuite_7000208867-miner2-1    |       go.vocdoni.io/dvote/db/pebbledb/pebledb.go:21 +0xb9
testsuite_7000208867-miner3-1    |       go.vocdoni.io/dvote/statedb/treeupdate.go:192 +0x110
testsuite_7000208867-miner2-1    |   go.vocdoni.io/dvote/db/pebbledb.(*WriteTx).Get()
testsuite_7000208867-miner3-1    |   go.vocdoni.io/dvote/statedb.(*treeUpdateView).DeepGet()
testsuite_7000208867-miner2-1    |       go.vocdoni.io/dvote/db/pebbledb/pebledb.go:69 +0xa7
testsuite_7000208867-miner3-1    |       go.vocdoni.io/dvote/statedb/treeupdate.go:378 +0xd2
testsuite_7000208867-miner2-1    |   go.vocdoni.io/dvote/db/prefixeddb.(*PrefixedWriteTx).Get()
testsuite_7000208867-miner3-1    |   go.vocdoni.io/dvote/vochain/state.getProcess()
testsuite_7000208867-miner2-1    |       go.vocdoni.io/dvote/db/prefixeddb/prefixeddb.go:117 +0x227
testsuite_7000208867-miner3-1    |       go.vocdoni.io/dvote/vochain/state/process.go:86 +0x1ae
testsuite_7000208867-miner2-1    |   go.vocdoni.io/dvote/tree/arbo.(*Tree).down()
testsuite_7000208867-miner3-1    |   go.vocdoni.io/dvote/vochain/state.(*State).Process()
testsuite_7000208867-miner2-1    |       go.vocdoni.io/dvote/tree/arbo/navigate.go:24 +0x1db
testsuite_7000208867-miner3-1    |       go.vocdoni.io/dvote/vochain/state/process.go:106 +0x27b
testsuite_7000208867-miner2-1    |   go.vocdoni.io/dvote/tree/arbo.(*Tree).down()
testsuite_7000208867-miner3-1    |   go.vocdoni.io/dvote/vochain/keykeeper.(*KeyKeeper).checkRevealProcess()
testsuite_7000208867-miner2-1    |       go.vocdoni.io/dvote/tree/arbo/navigate.go:77 +0x824
testsuite_7000208867-miner3-1    |       go.vocdoni.io/dvote/vochain/keykeeper/keykeeper.go:371 +0x608
testsuite_7000208867-miner2-1    |   go.vocdoni.io/dvote/tree/arbo.(*Tree).down()
testsuite_7000208867-miner3-1    |   go.vocdoni.io/dvote/vochain/keykeeper.(*KeyKeeper).Commit.func1()
testsuite_7000208867-miner2-1    |       go.vocdoni.io/dvote/tree/arbo/navigate.go:82 +0xc64
testsuite_7000208867-miner3-1    |       go.vocdoni.io/dvote/vochain/keykeeper/keykeeper.go:243 +0x3e
testsuite_7000208867-miner2-1    |   go.vocdoni.io/dvote/tree/arbo.(*Tree).GetWithTx()
testsuite_7000208867-miner3-1    | 
testsuite_7000208867-miner2-1    |       go.vocdoni.io/dvote/tree/arbo/tree.go:541 +0x789
testsuite_7000208867-miner3-1    | Previous write at 0x00c00a2e4000 by goroutine 255:
testsuite_7000208867-miner2-1    |   go.vocdoni.io/dvote/tree.(*Tree).Get()
testsuite_7000208867-miner2-1    |       go.vocdoni.io/dvote/tree/tree.go:70 +0x144
```

https://github.com/vocdoni/vocdoni-node/actions/runs/7000208867/job/19040533596#step:4:8191

File: cmd/end2endtest/account.go [hunk 3]
```diff
@@ -169,73 +169,121 @@ func testSendTokens(api *apiclient.HTTPclient, aliceKeys, bobKeys *ethereum.Sign
...
+		}
+		if err := checkAccountNonceAndBalance(bob, bobAcc.Nonce+2,
+			bobAcc.Balance-amountBtoA-(2*txCost+1)+amountAtoB); err != nil {
+			errorChan <- fmt.Errorf("failed when checkAccountNonceAndBalance for bob: %v", err)
+			return
+		}
+		wgCheck.Done()
+	}()
+
+	go func() {
```
Comment (altergui) [inline cmd/end2endtest/account.go:268]: you're missing a `wg.Add(1)` here

Comment (altergui) [reply cmd/end2endtest/account.go:268]: this is the cause of the panic

Comment (mariajdab) [reply cmd/end2endtest/account.go:268]: Yeah, my bad I forgot to commit that

**Review by altergui — APPROVED** [review]

**Pull request merged**